### DISCUSSION TITLE
Leaving driver open with multiple tabs

### DISCUSSION THREAD

**Discussion by invokeinnovation** [seleniumbase/SeleniumBase#2702] [Q&A] [answered]
How could I navigate to website attempt login, open new tab navigate to gmail etc, and switch back to previous tab once I've grabbed the access code?

Also is there a way to run multiple tabs at once and leave them all open each performing different task at a time?

this is the implementation I had 

for i in range(num_drivers):
    username, password = credentials[i]  # Get username and password for the current driver
    random_message = random.choice(messages)

    with SB(uc=True) as sb:
        url = url
        sb.driver.uc_open_with_reconnect(url, 2)

        # click the cookies accept button
        try:
            sb.click('//*[@id="app"]/span/div/div[3]/button[1]') 
        except:
            print("no cookies accept button")

        sb.type('#message-input', random_message+'\n')

        # sb.click('//*[@id="login-button"]') not present when going to chatroom
        sb.press_keys('//input[@name="username" and @id="input-element"]', username)
        sleep(1)
        sb.press_keys('//input[@name="password" and @id="input-element"]', password)
        sleep(1)
        sb.click('//*[@id="signin-modal"]/div/div[1]/div[2]/form/div/div/form/button')
        sleep(1)
        

        # implement code to get verification code, such as logging in via Gmail and extracting the code.
        try:
            sb.assert_text("Please", "div.my-8.font-semibold")
            print("login needs verfication")

        except:
            print("email verification not passed")

**Comment by mdmintz** [comment, marked as answer]
See the example for opening new tabs and switching between them: [SeleniumBase/examples/test_window_switching.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_window_switching.py)

For your format, you would be using `sb` instead of `self`.

Also note that some things with multiple tabs may work slightly differently in UC Mode, which may close the current tab and open a new one when navigating to a URL that has bot-detection services running on it.

**Reply by invokeinnovation** [reply]
okay thank you! Is there a way to keep the tab from closing ?

**Reply by mdmintz** [reply]
In UC Mode, you can use `sb.driver.uc_open(url)` to open a new page in the same window, but that might make you detected by anti-bot systems.
